iceoryx_posh/source/roudi/service_registry.cpp

```diff
@@ -118,6 +118,11 @@ void ServiceRegistry::find(const cxx::optional<capro::IdString_t>& service,
                            const cxx::optional<capro::IdString_t>& event,
                            cxx::function_ref<void(const ServiceDescriptionEntry&)> callable) const noexcept
 {
+    if (!callable)
```

Comment:
The more I think about it: it is bad for `function_ref` to be `nullable`, since 
1.  it semantically mimics a reference which is always referencing something (can be dangling, but this is the case here as well)
2. it leads to design flaws like these checks which add nothing but costs and boilerplate

Since its main purpose is to be passed around this is counter productive.

The situation is different for `std::function`, which is supposed to be stored and there it can be useful to default initialize it to something, and that something is naturally ... nothing.

Reply:
I see your point, but I'm not sure if we want to go in this strict and limiting direction for `iceoryx_hoofs`, see [my comment](https://github.com/eclipse-iceoryx/iceoryx/issues/1104#issuecomment-1040083614).